src/60_Samples_%26_Templates/Hotel.html

```diff
@@ -17,279 +18,149 @@
+    .rooms [role=listitem] > * {
+      margin-right: 16px
+    }
+    .rooms [role=listitem] button {
+      margin-left: auto;
+    }
+</style>
+</head>
+<body>
+  <h1 class="m1">
+    <amp-fit-text width="300" height="45" layout="responsive" max-font-size="45">
+      The Ampsterdam
+    </amp-fit-text>
+  </h1>
+  <p class="m1">The fastest hotel in town</p>
+  <h2 class="m1">When do you want to travel?</h2>
+  <!-- ## Date availability search -->
+  <!--
+  You can implement a date availability search by using the `amp-date-picker` as input of a form.
+  Here we are using `type=range` as we need 2 dates: the arriving and leaving date of an hotel reservation.
+  Upon receiving a `submit-succes` event from the form response, we set the value of a global variable `rooms`
+  to the form response.
+  -->
+  <form method="GET"
+        action="https://amp-by-example-live.glitch.me/rooms"
+        action-xhr="https://amp-by-example-live.glitch.me/rooms"
+        target="_top"
+        on="submit-success:AMP.setState({
+              rooms: event.response
+          })">
+      <amp-date-picker type="range"
+                        locale="en"
+                        format="Y-MM-DD"
+                        show-clear-dates
+                        min="2017-10-26"
+                        month-format="MMM"
+                        class="pt1 ml1">
+          <input amp-date-placeholder-start placeholder="arriving" id="arriving"/>
+          <input amp-date-placeholder-end placeholder="leaving" id="leaving"/>
+      </amp-date-picker>
+      <button class="ampstart-btn caps m1">Check Availability</button>
+    </form>
+
+    <!-- ## Additional filter -->
+    <!--
+    We implemented a filter to show only pet friendly rooms by adding a checkbox and
+    setting a new variable, `filterPetFriendly`, to the value of the filter.
+    -->
+    <label class="ml1">Show only pet friendly rooms: 
+      <input type="checkbox" class="m1" on="change:AMP.setState({
+        showPetFriendlyOnly: event.checked
+      })">
+    </label>
+  <!-- ## Filter function -->
+  <!--
+    We use `amp-bind-macro` to implement a function which filter the `amp-list` content
+    based on the pet friendly selection filter. By using `amp-bind-macro` we avoid repeating
+    the filter expression multiple times across the page.
+  -->
+  <amp-bind-macro id="filteredRooms"
+    expression="rooms.filter(room => showPetFriendlyOnly ? room.petFriendly : true)">
+  </amp-bind-macro>
+
+  <!-- ## Initial State -->
+  <!--
+    We are using `amp-state` to set the initial value of the variable `rooms` to the list of rooms
+    returned by the server. 
+  -->
+  <amp-state id="rooms" src="https://amp-by-example-live.glitch.me/rooms"></amp-state>
+
+  <!-- ## Rooms list -->
+  <!--
+    We are using `amp-list` to show a list of rooms. We bind the value of the `src` variable
+    to the result of the function `filterRooms` so that we can show the resulting list after the user selection.
```

Comment:
/resulting list after the user selection/available rooms based on the user selection.

Reply:
done